src/datapoints-graph/charts/charts.component.ts

```diff
@@ -224,14 +409,119 @@ export class ChartsComponent implements OnChanges, OnInit, OnDestroy {
+          series.data,
+          XAxisValue
+        );
+        if (!minValue) {
+          return;
+        }
+        const maxSeries = allDataPointSeries.find(
+          (s) => s.id === series.id.replace('/min', '/max')
+        );
+        const maxValue = this.findValueForExactOrEarlierTimestamp(
+          maxSeries.data as SeriesValue[],
+          XAxisValue
+        );
+        value =
+          `${minValue[1]} — ${maxValue[1]}` +
+          (series.datapointUnit ? ` ${series.datapointUnit}` : '') +
+          `<div style="font-size: 11px">${this.datePipe.transform(
+            minValue[0]
+          )}</div>`;
+      } else if (series.id.endsWith('/max')) {
+        // do nothing, value is handled  in 'min' case
+        return;
+      } else {
+        const seriesValue = this.findValueForExactOrEarlierTimestamp(
+          series.data,
+          XAxisValue
+        );
+        if (!seriesValue) {
+          return;
+        }
+        value =
+          seriesValue[1]?.toString() +
+          (series.datapointUnit ? ` ${series.datapointUnit}` : '') +
+          `<div style="font-size: 11px">${this.datePipe.transform(
+            seriesValue[0]
+          )}</div>`;
+      }
+
+      YAxisReadings.push(
+        `<span style='display: inline-block; background-color: ${series.itemStyle.color} ; height: 12px; width: 12px; border-radius: 50%; margin-right: 4px;'></span>` + // color circle
+          `<strong>${series.datapointLabel}: </strong>` + // name
+          value // single value or min-max range
+      );
+    });
+
+    const event = this.events.find((e) => e.type === params.data.itemType);
+    const alarm = this.alarms.find((a) => a.type === params.data.itemType);
+
+    let value: string;
+    if (event) {
+      // Add the event information to the value
+      value = `<div style="font-size: 11px">Event Time: ${event.time}</div>`;
+      value += `<div style="font-size: 11px">Event Type: ${event.type}</div>`;
+      value += `<div style="font-size: 11px">Event Text: ${event.text}</div>`;
+      value += `<div style="font-size: 11px">Event Last Updated: ${event.lastUpdated}</div>`;
+    }
+
+    if (alarm) {
+      // Add the alarm information to the value
+      value = `<div style="font-size: 11px">Alarm Time: ${alarm.time}</div>`;
+      value += `<div style="font-size: 11px">Alarm Type: ${alarm.type}</div>`;
+      value += `<div style="font-size: 11px">Alarm Text: ${alarm.text}</div>`;
+      value += `<div style="font-size: 11px">Alarm Last Updated: ${alarm.lastUpdated}</div>`;
+      value += `<div style="font-size: 11px">Alarm Count: ${alarm.count}</div>`;
+    }
+    YAxisReadings.push(value);
+    return (
+      this.datePipe.transform(XAxisValue) + '<br/>' + YAxisReadings.join('')
+    );
+  }
+
+  private async loadAlarmsAndEvents(): Promise<any> {
+    const timeRange = this.getTimeRange();
+    const filteredAlarmsOrEvents = this.config.alarmsEventsConfigs.filter(
```

Comment:
there is a good reason to filter this array, so
```suggestion
    const visibleAlarmsOrEvents = this.config.alarmsEventsConfigs.filter(
```